### DISCUSSION TITLE
Closing polygon vertice coordinates not identical to starting vertice coordinates?

### DISCUSSION THREAD

**Discussion by rexrobichaux** [geoman-io/leaflet-geoman#867] [Questions & Help] [answered]
Hello Geoman gurus!

 I started playing with geoman today in the hopes of creating a lightweight geometry editor that can pass coordinate values to our enterprise application. That application has the ability to write geometries (via lat/lon converted to st_geometry shapes) into our Esri (Oracle) enterprise geodatabase. 

I believe that Esri ST_geometry geodatabase shape validation requires input coordinates for polygons to have identical starting and closing/final vertice coordinates. Some examples here: https://desktop.arcgis.com/en/arcmap/latest/analyze/python/reading-geometries.htm

From some quick testing of the geoman editor and writing the coordinates to console, it appears to not contain a closing/identical coordinate pair for polygons. Is this expected behavior and can you think of anyway around this behavior aside from trying to build logic to create a duplicate origin polygon vertice and append that as the final coordinate pair upon completion of a polygon geometry? 

Here's an example console log of a simple polygon which in geoman was truly 5 vertices, but Esri would require a 6th (identical to #1 vertice) to be considered a valid shape for geodatabase storage: 

_latlngs: Array(1)
0: Array(5)
0: D {lat: 37.18657859524883, lng: -77.8271484375}
1: D {lat: 37.61423141542417, lng: -77.61840820312501}
2: D {lat: 37.666429212090605, lng: -77.06909179687501}
3: D {lat: 37.142803443716836, lng: -77.08007812500001}
4: D {lat: 37.00255267215955, lng: -77.56347656250001}
length: 5

Thanks for any ideas / suggestions you can provide!

-Rex

**Comment by Falke-Design** [comment, marked as answer]
Hi Rex,

with `_latlngs` you are useing the LatLng definition of Leaflet. (Same but public method would be `.getLatLngs()`). In Leaflet the first and last positions are **not** equivalent.
But you can convert it to geojson with `.toGeoJSON()`

GeoJSON definition:
> For type "Polygon", the "coordinates" member must be an array of LinearRing coordinate arrays.

[(2.1.6 Polygon)](https://geojson.org/geojson-spec.html#polygon)

> A LinearRing is closed LineString with 4 or more positions. The first and last positions are equivalent (they represent equivalent points).

[(2.1.4. LineString)](http://geojson.org/geojson-spec.html#linestring)


The default export type for Leaflet is GeoJSON, so there is also a way to import it with `L.geoJSON()`


Example:

Export:
```
var poly = L.polygon([yourCoords}).addTo(map);
var geojson = poly.toGeoJSON();
console.log(geojson);
```

Import:
```
L.geoJSON(geojson).addTo(map);
```

**Reply by rexrobichaux** [reply]
This is perfect @Falke-Design, and it makes sense that it's using the LatLng definition of Leaflet. Thank you for providing this geojson method as a workaround!
